Doc/library/fractions.rst

```diff
@@ -207,18 +207,34 @@ another rational number, or from a string.
 
       Provides support for formatting of :class:`Fraction` instances via the
       :meth:`str.format` method, the :func:`format` built-in function, or
-      :ref:`Formatted string literals <f-strings>`. The presentation types
-      ``"d"``, ``"e"``, ``"E"``, ``"f"``, ``"F"``, ``"g"``, ``"G"`` and ``"%"``
-      are supported. For presentation types other than ``"d"``, formatting for
-      :class:`Fraction` follows the rules outlined for the :class:`float` type
-      in the :ref:`formatspec` section. For presentation type ``"d"``,
-      formatting follows the rules for the :class:`int` type, except that the
-      zero-fill flag is not supported. If no presentation type is given, the
-      rules are identical to those for presentation type ``"d"``.
+      :ref:`Formatted string literals <f-strings>`.
+
+      If the ``format_spec`` format specification string does not end with one
+      of the presentation types ``'e'``, ``'E'``, ``'f'``, ``'F'``, ``'g'``,
+      ``'G'`` or ``'%'`` then formatting follows the general rules for fill,
+      alignment, sign handling, minimum width, and grouping as described in the
+      :ref:`format specification mini-language <formatspec>`. The "alternate
+      form" flag ``'#'`` is supported: if present, it forces the output string to
+      always include an explicit denominator, even when the value being
+      formatted is an exact integer. The zero-fill flag ``'0'`` is not
+      supported.
+
+      If the ``format_spec`` format specification string does end with one of
```

Comment:
"does end" or simply "ends"?

Reply:
Either is fine, grammatically; the "does end" wording works best in combination with the previous "does not end" paragraph, but works less well by itself. I'll change it.